core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -3607,7 +3607,7 @@ private void setAttributeSingleFile(RpcContext rpcContext, LockedInodePath inode
       if (ttl != null && inode.getTtl() != ttl) {
         entry.setTtl(ttl);
       }
-      if (inode.getTtlAction() != action) {
+      if (action != null && inode.getTtlAction() != action) {
```

Comment:
I think the check on line 3614 should also have the other two conditions (`get* != *`)?

Reply:
The last mod time should only ever be updated if the entry was actually modified. That behavior is consistent with the way `ext4` behaves on my CentOS 6 machine. Pushed a fix in f2a5fcf